outlines/generate/api.py

```diff
@@ -194,7 +192,9 @@ def __call__(
 
         stop_sequences = stop_at or self.stop_sequences
         max_tokens = max_tokens or self.max_tokens
+
         num_sequences = len(prompts)
+        fsms = [self.fsm.copy() for _ in prompts]
```

Comment:
Do we still need an array of FSMs now that they’re stateless?

Reply:
Yes, because of `CFGFSM` (which needs to track incremental parser state for each). That still needs to deepcopy new objects, but the others (`RegexFSM` and `StopAtFSM`) are stateless and can just return pointers to original object.